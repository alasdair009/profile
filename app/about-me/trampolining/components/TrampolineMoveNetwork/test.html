<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
