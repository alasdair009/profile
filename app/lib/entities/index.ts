"use client";

import styled from "styled-components";

export * from "@/entities";
export { Splash } from "./Splash";

export const FireWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: absolute;
  width: 100%;
`;
export const FirePadding = styled.div<{ $backgroundColor: string }>`
  background: ${({ $backgroundColor }) => $backgroundColor};
  flex: 1;
`;
