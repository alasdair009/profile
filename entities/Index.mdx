import { Meta } from "@storybook/blocks";
import Image from "next/image";
import amLogo from "./assets/am.svg";
import { rem } from "polished";

<Meta title="Alasdair Macrae Design" />

# Alasdair Macrae Design

<Image
  src={amLogo}
  alt="Green and white AM letters"
  width={0}
  height={0}
  style={{ maxWidth: rem(240), width: "100%", height: "auto" }}
/>

I am a keen supported of Storybook js using it across the majority of projects I build from websites, web-apps and even desktop applications.
