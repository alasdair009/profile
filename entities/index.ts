export { borderRadii } from "./design-tokens/effects";
export { colors } from "./design-tokens/colors";
export { fonts } from "./design-tokens/typography";
export { sizes } from "./design-tokens/dimensions";

export { Button } from "./atoms/Button";
export { Fire } from "./atoms/Fire";
export { Heading } from "./atoms/Heading";
export { Link } from "./atoms/Link";
export { Orb } from "./atoms/Orb";
export { Paragraph } from "./atoms/Paragraph";
export { Spacer } from "./atoms/Spacer";
export { TextMask } from "./atoms/TextMask";

export { Container } from "./organisms/Container";
export { ContentPlate } from "./organisms/ContentPlate";
export { Header } from "./organisms/Header";
export { Splash } from "./organisms/Splash";
