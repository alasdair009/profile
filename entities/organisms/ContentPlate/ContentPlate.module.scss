@use "../../../app/styles/animation";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";

$flipPoint: map-get(dimensions.$device, medium);
$turnAmount: 0.05;

@keyframes threeDRotate {
  0%,
  100% {
    transform: perspective(1000px) rotateX(-#{$turnAmount}turn) rotateY(0);
  }
  25% {
    transform: perspective(1000px) rotateX(0) rotateY(-#{$turnAmount}turn);
  }
  50% {
    transform: perspective(1000px) rotateX(#{$turnAmount}turn) rotateY(0);
  }
  75% {
    transform: perspective(1000px) rotateX(0) rotateY(#{$turnAmount}turn);
  }
}

.root {
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalDecorationMaxWidth);
  overflow: hidden;
  position: relative;
}

.inner {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  position: relative;
  width: 100%;

  @media ($flipPoint) {
    flex-direction: var(--orientation);
  }
}

.backgroundWrapper {
  background: var(--background);
  height: 100%;
  left: 0;
  margin: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

.foregroundWrapper {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 0;
  overflow: hidden;
  padding: dimensions.getSize(s16);
  position: relative;
  width: 100%;

  @media ($flipPoint) {
    padding: 0;
    width: 50%;
  }

  &.animate {
    animation: threeDRotate animation.animationDuration(3) infinite
      map-get(animation.$curves, linear);
  }
}

.foregroundImage {
  display: block;
  height: rem.toRem(400);
  max-height: 100%;
  max-width: 100%;
  object-fit: contain;
  width: rem.toRem(400);
}

.frameBoxWrapper {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: dimensions.getSize(s16);
  width: 100%;

  @media ($flipPoint) {
    width: 50%;
  }
}

.copyBox {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: dimensions.getSize(s16);
  width: 100%;

  @media ($flipPoint) {
    width: 50%;
  }
}
