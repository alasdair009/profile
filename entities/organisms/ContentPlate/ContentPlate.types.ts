import { Property } from "csstype";
import { StaticImageData } from "next/image";
import { HTMLAttributes, ReactNode } from "react";

export type ContentPlateProps = {
  orientation?: "left" | "right";
  backgroundCss?: Property.Background;
  backgroundImage?: StaticImageData;
  backgroundImageAlt?: string;
  foregroundImage?: StaticImageData;
  foregroundImageAlt?: string;
  children: ReactNode;
} & HTMLAttributes<HTMLDivElement>;
