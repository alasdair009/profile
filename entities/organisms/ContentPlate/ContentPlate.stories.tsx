import { ContentPlate } from "./ContentPlate";
import { Meta, StoryObj } from "@storybook/react";
import { colors, Flame, Heading, Paragraph } from "@/entities";
import amLogo from "../../assets/am.svg";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof ContentPlate> = {
  component: ContentPlate,
  argTypes: {
    backgroundCss: {
      control: {
        type: "color",
        // presetColors: Object.values(colors),
      },
    },
    children: {
      type: "string",
    },
  },
  args: {
    backgroundCss: colors.blackEvil,
    children: (
      <>
        <Heading level="h3" data-testid={`${ContentPlate.name}Heading`}>
          Children
        </Heading>
        <Paragraph data-testid={`${ContentPlate.name}Paragraph`}>
          This is my content plate, there are many like it but this one is mine!
        </Paragraph>
      </>
    ),
  },
};
export default meta;

export const Default: StoryObj<typeof ContentPlate> = {
  args: {
    foregroundImage: amLogo,
    foregroundImageAlt: "AM Logo",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const contentPlateElement: HTMLDivElement = canvas.getByTestId(
      ContentPlate.name
    );

    await expect(contentPlateElement).toBeInTheDocument();
    await expect(
      canvas.getByAltText(args.foregroundImageAlt as string)
    ).toBeInTheDocument();
    await expect(contentPlateElement.children[0]).toHaveStyle(
      `background-color: ${args.backgroundCss}`
    );
  },
};

export const WithFlame: StoryObj<typeof ContentPlate> = {
  args: {
    flameColor: "red",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const contentPlateElement: HTMLDivElement = canvas.getByTestId(
      ContentPlate.name
    );

    await expect(contentPlateElement).toBeInTheDocument();
    await expect(canvas.getByTestId(Flame.name)).toBeInTheDocument();
  },
};
