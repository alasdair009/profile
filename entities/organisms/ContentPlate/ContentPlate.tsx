import {
  Root,
  Inner,
  BackgroundWrapper,
  ForegroundWrapper,
  CopyBox,
} from "./styles";
import Image from "next/image";
import { ContentPlateProps } from "./ContentPlate.types";

export function ContentPlate({
  orientation = "left",
  backgroundCss = "none",
  backgroundImage,
  backgroundImageAlt,
  foregroundImage,
  foregroundImageAlt,
  children,
  ...rest
}: ContentPlateProps) {
  return (
    <Root {...rest} data-testid={ContentPlate.name}>
      {((backgroundImage && backgroundImageAlt) || backgroundCss) && (
        <BackgroundWrapper $backgroundCss={backgroundCss}>
          {backgroundImage && backgroundImageAlt && (
            <Image src={backgroundImage} alt={backgroundImageAlt} />
          )}
        </BackgroundWrapper>
      )}
      <Inner $orientation={orientation}>
        <CopyBox>{children}</CopyBox>
        {foregroundImage && foregroundImageAlt && (
          <ForegroundWrapper>
            <Image
              height={400}
              width={400}
              src={foregroundImage}
              alt={foregroundImageAlt}
            />
          </ForegroundWrapper>
        )}
      </Inner>
    </Root>
  );
}
