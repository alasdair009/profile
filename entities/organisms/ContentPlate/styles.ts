"use client";
import styled from "styled-components";
import { Property } from "csstype";
import {
  globalContentMaxWidth,
  globalDecorationMaxWidth, sizes,
} from "@/entities/design-tokens/dimensions";
import { rem } from "polished";
import { ContentPlateProps } from "@/entities/organisms/ContentPlate/ContentPlate.types";

export const Root = styled.section`
  margin: 0 auto;
  max-width: ${rem(globalDecorationMaxWidth)};
  overflow: hidden;
  position: relative;
`;

export const Inner = styled.div<{
  $orientation: Exclude<ContentPlateProps["orientation"], undefined>;
}>`
  display: flex;
  flex-direction: ${({ $orientation }) => $orientation};
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  position: relative;
  width: 100%;
`;

export const BackgroundWrapper = styled.figure<{
  $backgroundCss: Property.Background;
}>`
  background: ${({ $backgroundCss }) => $backgroundCss};
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
`;

export const ForegroundWrapper = styled.figure`
  align-items: center;
  display: flex;
  justify-content: center;
  overflow: hidden;
  position: relative;
  width: 50%;
`;

export const CopyBox = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: ${sizes.s16.rem};
  width: 50%
`;
