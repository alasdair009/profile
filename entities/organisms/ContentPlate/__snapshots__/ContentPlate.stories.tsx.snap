// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/ContentPlate Default smoke-test 1`] = `
<section data-testid="ContentPlate"
         class="sc-dKsqdn ldIjne"
>
  <figure class="sc-iAlELC jqxhvc">
  </figure>
  <div class="sc-eowDPD kdYnoU">
    <div class="sc-hDcvty pLorZ">
      <h3 data-testid="Heading"
          class="sc-hfvVTD cmyMfV"
      >
        Children
      </h3>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        This is my content plate, there are many like it but this one is mine!
      </p>
    </div>
    <figure class="sc-oeqTF bFmTDs">
      <img alt="AM Logo"
           loading="lazy"
           width="176"
           height="146"
           decoding="async"
           data-nimg="1"
           class="sc-ezTrPE wzeXg"
           srcset="/static/media/entities/assets/am.svg?w=256&amp;q=75 1x, /static/media/entities/assets/am.svg?w=384&amp;q=75 2x"
           src="/static/media/entities/assets/am.svg?w=384&amp;q=75"
           style="color: transparent;"
      >
    </figure>
  </div>
</section>
`;
